expressions/table_generator/GeneratorFunctionFactory.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ *   Copyright 2016 Pivotal Software, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_EXPRESSIONS_TABLE_GENERATOR_GENERATOR_FUNCTION_FACTORY_HPP_
+#define QUICKSTEP_EXPRESSIONS_TABLE_GENERATOR_GENERATOR_FUNCTION_FACTORY_HPP_
+
+#include <string>
+
+#include "expressions/table_generator/GeneratorFunction.pb.h"
+#include "utility/Macros.hpp"
+
+namespace quickstep {
+
+class GeneratorFunction;
+class GeneratorFunctionHandle;
+
+/** \addtogroup Expressions
+ *  @{
+ */
+
+/**
+ * @brief All-static factory class that provides access to the various concrete
```

Comment:
Need a more detailed description for what the generator functions are -- just need to expand on the comments here.